falcon/inspect.py

```diff
@@ -753,7 +777,17 @@ def visit_app(self, app: AppInfo) -> str:
 # ------------------------------------------------------------------------
 
 
-def _get_source_info(obj, default='[unknown file]'):
+def _get_source_info(
+    obj: ModuleType
+    | Type[Any]
+    | MethodType
+    | FunctionType
+    | TracebackType
+    | FrameType
+    | CodeType
+    | Callable[..., Any],
+    default: Optional[str] = '[unknown file]',
+) -> Optional[str]:
```

Comment:
default and the return type are actually `str`, not optional

this would avoid the need to add the assert above

Reply:
Yes, but there is a case in the tests where default is None, so it is possible that this function will return it.

Reply:
interesting. I'll take a look, since I think it's not correct (from a quick look everywhere it's used it assumes that it returns a str not None

Reply:
Moved to typing